src/controllers/hid/hidcontroller.cpp

```diff
@@ -101,18 +100,36 @@ int HidController::open() {
     // Set hid controller to non-blocking
     if (hid_set_nonblocking(m_pHidDevice, 1) != 0) {
         qCWarning(m_logBase) << "Unable to set HID device " << getName() << " to non-blocking";
+        hid_close(m_pHidDevice);
         return -1;
     }
 
-    // This isn't strictly necessary but is good practice.
-    for (int i = 0; i < kNumBuffers; i++) {
-        memset(m_pPollData[i], 0, kBufferSize);
-    }
-    m_lastPollSize = 0;
-
     setOpen(true);
+
+    m_pHidIoThread = std::make_unique<HidIoThread>(m_pHidDevice, std::move(m_deviceInfo));
+    m_pHidIoThread->setObjectName(QString("HidIoThread %1").arg(getName()));
+
+    connect(m_pHidIoThread.get(),
+            &HidIoThread::receive,
+            this,
+            &HidController::receive,
+            Qt::QueuedConnection);
+
+    connect(this,
+            &HidController::sendOutputReport,
+            m_pHidIoThread.get(),
+            &HidIoThread::sendOutputReport,
+            Qt::QueuedConnection);
```

Comment:
I changed the implementation for OutputReports. Instead of the QueuedConnection, I use an buffer/latch per OutputReport now, where the code has full control. In overload cases it skips superseded OutpuReports now.